### PR TITLE
Feat/pypi ci

### PR DESCRIPTION
**Pull request by zhenfeng-zhu**
### Description

This is for issue https://github.com/aptos-labs/aptos-core/issues/7351

### CODE REVIEW

**Review by gedigi — APPROVED** [review]

File: .github/workflows/python_sdk_publish.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
```
Comment (banool) [inline .github/workflows/python_sdk_publish.yaml:3]: I don't think our current Python SDK testing process is strict enough for this to run on every push right now. Could we make this work on workflow_dispatch instead for now?

File: .github/workflows/python_sdk_publish.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: Build project for distribution
+        run: cd ./ecosystem/python/sdk && poetry build
+
+      - name: Check Version
+        id: check-version
+        run: |
+          [[ "$(poetry version --short)" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || echo prerelease=true >> $GITHUB_OUTPUT
+
```
Comment (banool) [inline .github/workflows/python_sdk_publish.yaml:33]: Ideally we would also check that there is an entry in the CHANGELOG for this new version.

**Review by banool — CHANGES_REQUESTED** [review]
Thanks for the PR! Requesting changes for the workflow trigger.

File: .github/workflows/python_sdk_publish.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install Poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python - -y
```
Comment (banool) [inline .github/workflows/python_sdk_publish.yaml:21]: Let's use `actions/python-setup/action.yaml` instead, it sets up the correct Python version as configured in the repo, installs Poetry, etc.

**Comment by zhenfeng-zhu** [comment]
Thanks for all your help, I have resolved some comment. Please review it!

File: .github/workflows/python-sdk-publish.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+name: "Run Python SDK Publish"
+
+on:
+  workflow_dispatch:
+  push:
```
Comment (rustielin) [inline .github/workflows/python-sdk-publish.yaml:5]: we don't need the `push` trigger, can we just do it for `workflow_dispatch`?

File: .github/workflows/python-sdk-publish.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          pyproject_directory: ./ecosystem/python/sdk
+
+      - name: Build project for distribution
+        run: poetry build
+        working-directory: ./ecosystem/python/sdk
+
+      - name: Check Version
+        id: check-version
+        run: |
+          [[ "$(poetry version --short)" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] && [[ $(grep $(poetry version --short) CHANGELOG.md) =~ $(poetry version --short)  ]] || echo prerelease=true >> $GITHUB_OUTPUT
```
Comment (rustielin) [inline .github/workflows/python-sdk-publish.yaml:26]: where is this github output being used?

**Review by rustielin — CHANGES_REQUESTED** [review]

**Review by banool — APPROVED** [review]
Looks good, thanks for changing to workflow_dispatch!!

**Review by sionescu — APPROVED** [review]

**Pull request merged**